librewolf: added extra config files to persist, and minor hash fix

brian6932 opened the pull request:
also fixed minor mistake in hash

<!-- Provide a general summary of your changes in the title above -->

Those config files/dirs are required by userchrome configurations, so it makes sense to persist them


- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

issaclin32 commented:
Thanks for the fix.
Could you provide some information/documentation about `default` and `config.js`?

*Scoop* uses the portable version of *LibreWolf*, and there is `(INSTALL DIRECTORY)\Librewolf\defaults`.
I am not sure whether you are trying to persist
**(1)** `(INSTALL DIRECTORY)\Librewolf\defaults` and `(INSTALL DIRECTORY)\Librewolf\config.js`
, or just **(2)** `(INSTALL DIRECTORY)\defaults` and `(INSTALL DIRECTORY)\config.js`

Also, since `config.js` does not exist in the archive, we need to create it first by:
```
"pre_install": "if (!(Test-Path \"$persist_dir\\config.js\")) { New-Item \"$dir\\config.js\" | Out-Null }"
```
Otherwise *Scoop* will create `config.js` as a folder by default.

brian6932 commented:
> Could you provide some information/documentation about `default` and `config.js`?

https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig

> I am not sure whether you are trying to persist
> **(1)** `(INSTALL DIRECTORY)\Librewolf\defaults` and `(INSTALL DIRECTORY)\Librewolf\config.js`
> , or just **(2)** `(INSTALL DIRECTORY)\defaults` and `(INSTALL DIRECTORY)\config.js`

Fixed, also I should've specified `defaults/prefs` as a nested dir

> Also, since `config.js` does not exist in the archive, we need to create it first by:
> 
> ```
> "pre_install": "if (!(Test-Path \"$persist_dir\\config.js\")) { New-Item \"$dir\\config.js\" | Out-Null }"
> ```
> 
> Otherwise _Scoop_ will create `config.js` as a folder by default.

Added, minor correction, it does not have to be `config.js`, any `*.js` file should work as long as the name is set as `general.config.filename` in a `*.js` file in `defaults/prefs`, is it possible to use a wildcard with persist?

issaclin32 commented on bucket/librewolf.json:
```diff
@@ -26,7 +27,11 @@
-    "persist": "Profiles",
+    "persist": [
+        "Profiles",
+        "LibreWolf\\defaults\\prefs",
+        "LibreWolf\\*.js"
```
```suggestion
        "LibreWolf\\defaults\\prefs"
```

**persist** does not support wildcards.
If order to save `*.js`, we need to copy it in `pre_install` and `pre_uninstall` like:

```
"pre_install": [
    "......",
    "if (Test-Path \"$dir\\Librewolf\\*.js\") { Copy-Item \"$dir\\Librewolf\\*.js\" \"$persist_dir\\\" }"
]
"pre_uninstall": "if (Test-Path \"$persist_dir\\Librewolf\\*.js\") { Copy-Item \"$persist_dir\\Librewolf\\*.js\" \"$dir\\\" }"
```

issaclin32 commented:
/verify

brian6932 commented:
Uhh, it seems like https://gitlab.com/librewolf-community/browser/windows/uploads/c5782ae771d4ded03bdfc2f4a3850264/sha256sums.txt literally does not exist in https://gitlab.com/api/v4/projects/13852981/releases, should hash verification be removed all together?

brian6932 commented:
ok nvm, I fixed it

brian6932 commented:
/verify

brian6932 commented:
/verify

brian6932 commented:
why not?

brian6932 commented:
oh the id is different

issaclin32 commented:
/verify

brian6932 commented:
so apparently the order of the id and sha changes in the array, your solution is best 👍

issaclin32 commented:
Looks fine now. Thank you for the PR.
